runtime/jextractnatives/jextractnatives.c

```diff
@@ -461,37 +455,38 @@ flushCache(void)
 /**
  * This function implements a very simple caching scheme to accelerate the reading of small objects.
  * A more sophisticated scheme is implemented in the Java code. This cache allows us to bypass
- * the relatively expensive call-in to Java for most objects. During a normal jextract run we expect 
+ * the relatively expensive call-in to Java for most objects. During a normal jextract run we expect
  * the cache to have a hit rate of over 90%.
  */
-static void 
+static void
 readCachedMemory(UDATA address, void *structure, UDATA size, UDATA *bytesRead)
 {
-	static UDATA hits, total;
+	static UDATA hits;
+	static UDATA total;
 
 	dbgCacheElement* thisElement = NULL;
 	UDATA lineStart = address & ~(UDATA)(sizeof(thisElement->data) - 1);
 	UDATA endAddress = address + size;
 
 #if 0
-	if ( ((++total) % (16 * 1024)) == 0) {
-		dbgPrint("Cache hit rate: %.2f\n", (float)hits / (float)(total));
+	if (((++total) % (16 * 1024)) == 0) {
```

Comment:
This is using uninitialized variables. Also `total` is unused outside of this disabled block of code.

Reply:
The C language guarantees that static variables are initialized to zero.
I'll make that explicit and make all the statistical stuff conditional.